numpy/core/_internal.py

```diff
@@ -287,23 +287,45 @@ def _newnames(datatype, order):
         return tuple(list(order) + nameslist)
     raise ValueError("unsupported order value: %s" % (order,))
 
+def isstring(obj):
```

Comment:
I would prefer a conditional def instead of a condition inside the function. Do we actually support unicode fields on python 2? If not, I think you can just use `str`, if we do, you can use `six.string_types` instead.